v-next/core/src/global-dir.ts

```diff
@@ -11,6 +11,21 @@ export async function getConfigDir(): Promise<string> {
   return config;
 }
 
+/**
+ * Returns the path to the telemetry directory for the specified package.
+ * If no package name is provided, the default package name "hardhat" is used.
+ * Ensures that the directory exists before returning the path.
+ *
+ * @param packageName - The name of the package to get the telemetry directory for. Defaults to "hardhat".
+ *
+ * @returns A promise that resolves to the path of the telemetry directory.
+ */
+export async function getTelemetryDir(packageName?: string): Promise<string> {
```

Comment:
Can we make sure that this dir is not deleted by the `clean --global` task? /cc @schaable

Reply:
I don't think so, as the `global` flag deletes the `cache` path and telemetry uses the `data` path.